include/blogPost.php

```diff
@@ -1,62 +1,52 @@
 <?php
-function getAllBlogPosts(){
-    $post = array(
-        array(
-            'postId'=>0,
-            'title'=> "Getting To Know Me <br/>",
-            'body'=> "<div class = 'aboutMe'>
-                    Below are some more facts about me: <br/> <br/>
-                    I am from NY <br/>
-                     My favorite dessert is ice cream <br/>
-                     My least favorite forrd is chocolate <br/>
-                     I am scared of cats <br/>
-             My favorite Disney movie is mulan <br/><br/><br/>"
-         ),
-         array(
-             'postId'=>1,
-             'title'=> "What I am Studying <br/>",
-             'body'=> "<div class = 'aboutMe'>
-                     I am currently studying CogNeuro<br/><br/><br/>",
-          ),
-           array(
-               'postId'=>2,
-               'title'=> "What I'm Listening to <br/>",
-               'body'=> "<div class = 'aboutMe'>
-                    Explore the album covers to see what I am listening to!<br/><br/><br/>",
-              ));
-          return $post;
-}
-// include_once('config/init.php');
-// function getAllBlogPostsPost($postID){
-// $Result = dbQuery("
-//     SELECT *
-//     FROM blog_post
-//     WHERE blogPostId =:blogPostId
-//     ", array("blogPostId"=>$postId));
-//     return $Result;
-// }
-
-function echoPostLinkHtml(){
-    $post = getAllBlogPosts();
-    for ($i = 0; $i<count($post); $i++){
-        $postId = $post[$i]['postId'];
-        $postTitle = $post[$i]['title'];
-        $postBody= $post[$i]['body'];
-      echo  "<a class = 'resume' href = 'viewPost.php?postId=". $postId."'>".$postTitle." </a><br/>";
+include_once('config/init.php');
+function blogTagOrganizer (){
+    $tagTypes = dbQuery("
+            SELECT *
+            FROM tags
+            ")->fetchALL();
+    $tagArray = array();
+        foreach ($tagTypes as $tagsOrganized){
+            if(!in_array($tagsOrganized['tagName'], $tagArray)){
+                array_push($tagArray, $tagsOrganized['tagName']);
+                echo "<a class = 'resume' href = 'listOfBlogs.php?tagPostId=". $tagsOrganized['tagId']."'>".$tagsOrganized['tagName']." </a>";
+                }
+            }
+        }
+function echoPostLinkHtml($postTagId){
+    $tagIdOrganizer = dbQuery("
+        SELECT * FROM blog_post
+        INNER JOIN blogPost_tags_linked ON blogPost_tags_linked.blogPostId = blog_post.blogPostId
+        INNER JOIN tags ON tags.tagId =  blogPost_tags_linked.tagId
+        WHERE tags.tagId = :varTagName
+        ", array("varTagName" => $postTagId))->fetchALL();
+        $tagPostArray = array();
+        foreach ($tagIdOrganizer as $tagPosts){
+            if(!in_array($tagPosts['tagName'], $tagPostArray)){
```

Comment:
Something to consider (you don't necessarily need to change this): this `in_array` function loops through all the values in an array to see if the one you're looking for exists. For such a small amount of data it's not a big deal, but this is generally frowned upon because it would perform poorly with large data sets. Instead, you can do this:
```
if(@$tagPostArray[$tagPosts['tagName']]){
    echo "<h4 class='projects'>".$tagPosts['tagName']."</h4>";
    $tagPostArray[$tagPosts['tagName']] = true;
}
```

This puts the tag name into the index of the array rather than the value, and it's easier to look that up. That probably doesn't make any sense without understanding how arrays work at a low level. You can ignore this if you want, but you seem to be grasping things really well so far, so if you want to dive deeper, ask me about this and I'll explain it more.